### DISCUSSION TITLE
Add optional "read" action to Activity Logs

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#12899] [Feature Requests]
Currently the only actions that are available in the activity logs are create, update and delete.

We're working on implementing Directus for a HIPAA regulated healthcare application where we need to be able to audit patient information access, see who accessed it, when, etc...

We can implement this manually by adding event hooks to <collection_name>.items.read events, but it would be a great feature to be able to optionally add the "read" events to the log automatically.

For individual item access, this is straightforward. For list access, not so much. In the list/search case, I think just logging the query (like with the read event) would work great. From that you can determine what was accessed for auditability.

I'm getting more and more into the source code for the project, so if this seems like a good idea, maybe I could work on a PR for it?

**Comment by rijkvanzanten** [comment]
Totally get the use case, but I'm a little worried about the implications of saving _every_ read into directus_activity. That could generate millions and millions of rows in directus_activity, which might cause other trouble 🤔 We would have to figure out some proper way to truncate that table efficiently (maybe write to a file after X days or something?) in order to prevent people from accidentally blow up the instance

**Reply by claytongulick** [reply]
For sure, that's why I was thinking about it being an optional toggle, perhaps even by collection. Some way to specify on a per-collection basis "these are the actions I want to audit on this collection". In my case, I would only turn it on for any collection that contained PHI.

I don't think it's necessary to store line-item access records for lists/searches - I think just the query would be fine. From a regulatory / access standpoint I always do "minimum necessary" in search results, so that's generally fairly low-risk category data. Wheras a full /patient/id retrieval would count as a patient record access event and would definitely need to be auditable. 

In real-world use, it's not that big of a deal on data volume, certainly not as bad as analytics telemetry. Pretty much every healthcare system I've ever built has this. It's also very useful for UX improvements for doing things like "recently accessed" shortcuts and "most popular items" etc...

**Reply by alexbrowng** [reply]
I like the idea of ​​being able to track the readings of some collections, regarding the problem that it can pose to the volume of items, it could be allowed to configure a life time policy, which automatically eliminates items that are older than X days or you could opt for a multi-table model by time periods, that is, activity_2022_01 for January items, activity_2022_02, in any case, I think that the two functionalities are important independently of each other

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
